### PR TITLE
Use postcss for better handling of css transformation

### LINKED ISSUE #18: rawCssNamespace does not resolve the content as sass
The new strategy for namespacing adds the wrapper as sass notation. I have only released now that styled components are not really sass. 

Given `rawCssNamespace: ['body .theme-light', '.theme-dark']` and styles:
```jsx
styled.div`
  display: block;
  min-width: 200px;

  &.myClass {
    /* some css */
  }
`
```
When the plugin namespace styles
Then it should not add namespace as wrapper

Current behaviour:
```jsx
styled.div`
  body .theme-light, .theme-dark {
    & {
      display: block;
      min-width: 200px;

      &.myClass {
        /* some css */
      }
    }
  }
`
```
Expected behaviour:
Current behaviour:
```
styled.div`
  body .theme-light &, .theme-dark & {
    display: block;
    min-width: 200px;
  }
  body .theme-light &.myClass, .theme-dark &.myClass  {
    /* some css */
  }
`
```

I will have a look at this weekend. I see to possible solutions:
* Introduce sass to parse it (which I like more);
* Create a module to analyse styles extracting the raw css as self and the rules scoped in its own module;

@nvanselow Sorry for that, it was good releasing as an alternative strategy. Let me know which solution you are up to, also, if you have a different solution. ;)

Thank you

### LINKED ISSUE #21: .extend`` not working with namespace
Hey there,
I am not entirely sure if this is an issue for intended behaviour, so maybe someone can just explain?!
But I have the problem, that I have some styled components, which I am also extending and then using in a bigger application. That's why I wanted to use this plugin, to up the specificity of those components.
For example
```js
export const Button = styled.button`
  [...]
`;

export const ExtendedButton = Button.extend`
  [...]
`;
```
The problem is that the extended styles don't seem to get the prefix like the main styles do:
screen shot 2018-09-06 at 13 10 18

Just for reference, this is the config for the plugin I am using: `["@quickbaseoss/babel-plugin-styled-components-css-namespace", { "cssNamespace": "folders" }]`

Can anyone help?

### LINKED ISSUE #22: Upgrade to Babel 7
Hi all!

Babel has updated to v7 and it would be great to update this package to follow suite with upgrading its dependencies.

### LINKED ISSUE #24: Fix Known Issue with Sibling Selector
Hi I'm using `styled-components` in my current project and want to use this babel plugin as suggested by the official docs to prevent third-party styling of my components when embedded as a widget.

The styled components in this projects make quite some use of the & and sibling operator ~ which is not working and a known issue as stated in the docs. Is there any ambition to fix this in a future release as it seems to be fixed when using the `rawCssNamepsace` option? Unfortunately the `rawCssNamespace` option does not work in my project and my `webpack dev server` is not starting, it seems to be stuck in an infinite loop. I tried to debug it but till now I had no luck. Any hints are highly appreciated.

### LINKED ISSUE #25: rawCssNamespace will add suffix and not prefix when there is no nested class in css
Hey!

When you have component like this:
```jsx
const Button = styled.button`
  background: green;
  color: red;
  padding: 1em;
`;
```

and .babelrc like:
```
["@quickbaseoss/babel-plugin-styled-components-css-namespace", {
  "rawCssNamespace": ["#prefix"]
}]
```

The generated selector is wrong:
```html
<style data-styled="" data-styled-version="4.1.3">
/* sc-component-id: sc-bdVaJa */
.gBOsWq #prefix{background:green;color:red;padding:1em;}</style>
```

The correct should be
```html
<style data-styled="" data-styled-version="4.1.3">
/* sc-component-id: sc-bdVaJa */
#prefix .gBOsWq {background:green;color:red;padding:1em;}</style>
```
Using the cssNamespace seems to work, but I need to prefix the selectors with id rather than class.

**Note**
If I change 
```jsx
const Button = styled.button`
  background: green;
  color: red;
  padding: 1em;
`;
```

to

```jsx
const Button = styled.button`
  .foo {
    background: green;
    color: red;
    padding: 1em;
  }
`;
```

It works.

Is this a desired behaviour?

### LINKED ISSUE #27: Update on progress
January is going to be the month of closing out some issues in this project. Unfortunately, I was unable to dedicate enough time to improvements here. Thank you to everyone who has reported issues. Please continue to report them or open PRs with fixes. My goal is to better be aware of issues coming into the project and responding more quickly going forward.

### PR DESCRIPTION
**Pull request by nvanselow**
A rethinking of the plugin that uses postcss to better handle the css transformation required to add the css namespacing.

Additionally, all the dependencies have been updated to latest.

I set the version to 1.0.0-rc so people can try this out and we can fix issue before  an official 1.0.0 release.

I believe this version of the plugin takes care of all the issues that were reported on the previous version.

Closes #18 
Closes #21 
Closes #22 
Closes #24 
Closes #25 
Closes #27

### CODE REVIEW

File: src/visitors/cssNamespace.js [hunk 1]
```diff
@@ -1,122 +1,144 @@
...
+      // Which even the safe parser doesn't know how to handle.
+      // So we add a fake value to make it valid css.
+      // .button {
+      //   EXPRESSION: fakevalue;
+      // }
+      if (
+        (rawValueWithoutWhiteSpace === '' ||
+          rawValueWithoutWhiteSpace.endsWith('{') ||
+          rawValueWithoutWhiteSpace.endsWith('}') ||
+          rawValueWithoutWhiteSpace.endsWith(';')) &&
```
Comment (nvanselow) [inline src/visitors/cssNamespace.js]: I don't love this, but I haven't found a better way to detect when an expression appears on its own line and causes invalid css.

**Comment by Pczek** [comment]
Would love to test this PR if it fixes my issue. It's not yet released as a RC right?

**Comment by nvanselow** [comment]
@Pczek Not yet. I may go ahead and merge and release it as an rc though so people can test.

File: src/visitors/cssNamespace.js (hunk 1 above)
Comment (TrevorBurnham) [reply src/visitors/cssNamespace.js]: I ended up hacking the parser to solve this: https://github.com/TrevorBurnham/babel-plugin-namespace-styled-components/pull/2

Comment (nvanselow) [reply src/visitors/cssNamespace.js]: Awesome. Thanks for the heads up. That might solve another issue that was opened as well.

**Pull request merged**